packages/blockchain-link/src/workers/solana/index.ts

```diff
@@ -0,0 +1,491 @@
+            txids: txIdPage,
+        },
+        page: transactionPage
+            ? {
+                  total: allTxIds.length,
+                  index: pageNumber,
+                  size: transactionPage.length,
+              }
+            : undefined,
+        tokens,
+        misc: {
+            owner: accountInfo.owner.toString(),
+        },
+    };
+    return {
+        type: RESPONSES.GET_ACCOUNT_INFO,
+        payload: account,
+    } as const;
+};
+
+const getInfo = async (request: Request<MessageTypes.GetInfo>) => {
+    const api = await request.connect();
+    const { blockhash: blockHash, lastValidBlockHeight: blockHeight } =
+        await api.getLatestBlockhash('finalized');
+    const isTestnet =
+        (await api.getGenesisHash()) !== '5eykt4UsFv8P8NJdTREpY1vzqKqZKvdpKuc147dw2N9d';
+
+    const serverInfo = {
+        // genesisHash is reliable identifier of the network, for mainnet the genesis hash is 5eykt4UsFv8P8NJdTREpY1vzqKqZKvdpKuc147dw2N9d
+        testnet: isTestnet,
+        blockHeight,
+        blockHash,
+        shortcut: isTestnet ? 'dsol' : 'sol',
+        url: api.rpcEndpoint,
+        name: 'Solana',
+        version: (await api.getVersion())['solana-core'],
+        decimals: 9,
+    };
+    return {
+        type: RESPONSES.GET_INFO,
+        payload: { ...serverInfo },
+    } as const;
+};
+
+const estimateFee = async (request: Request<MessageTypes.EstimateFee>) => {
+    const api = await request.connect();
+
+    const message = request.payload.specific?.data;
+    const isCreatingAccount = request.payload.specific?.isCreatingAccount;
+
+    if (message == null) {
+        throw new Error('Could not estimate fee for transaction.');
+    }
+
+    const result = await api.getFeeForMessage(Message.from(Buffer.from(message, 'hex')));
+
+    // The result can be null, for example if the transaction blockhash is invalid.
+    // In this case, we should fall back to the default fee.
+    if (result.value == null) {
+        throw new Error('Could not estimate fee for transaction.');
+    }
+
+    const accountCreationFee = isCreatingAccount
+        ? await api.getMinimumBalanceForRentExemption(TOKEN_ACCOUNT_LAYOUT.span)
+        : 0;
+
+    const payload = [
+        {
+            feePerUnit: `${result.value + accountCreationFee}`,
+        },
+    ];
+
+    return {
+        type: RESPONSES.ESTIMATE_FEE,
+        payload,
+    } as const;
+};
+
+const BLOCK_SUBSCRIBE_INTERVAL_MS = 10000;
+const subscribeBlock = async ({ state, connect, post }: Context) => {
+    if (state.getSubscription('block')) return;
+    const api = await connect();
+
+    // the solana RPC api has subscribe method, see here: https://www.quicknode.com/docs/solana/rootSubscribe
+    // but solana block height is updated so often that it slows down the whole application and overloads the the api
+    // so we instead use setInterval to check for new blocks every 10 seconds
+    const interval = setInterval(async () => {
```

Comment:
note: make sure that we don't register some unwanted intervals in case or repeated calls to this. maybe intervals should be cleared sometimes.

Reply:
cc: @PeterBenc

Reply:
Yes @mroz22 should be solved by line 280, if there is a subscription already, we don't create another one. Also line 310 clears the interval. Do you consider this sufficient?